resource-managers/yarn/pom.xml

```diff
@@ -126,7 +126,7 @@
     </dependency>
 
      <!--
-    See SPARK-3710. hadoop-yarn-server-tests in Hadoop 2.2 fails to pull some needed
```

Comment:
I tried removing this dependency and it removed a mock YARN class that the test was depending on, so it seemed necessary to keep the hadoop-yarn-server-tests even in 2.6. As for the dependencies, I didn't try, but let me just see what happens if I remove those two